### ISSUE TITLE
Missing "Reexport" dependency

### ISSUE THREAD

**Issue by swiesend** [sglyon/PlotlyJS.jl#215]
Is this an issue or just me?

```
[ Info: Precompiling PlotlyJS [f0f68f2c-4968-5e81-91da-67840de0976a]
ERROR: LoadError: ArgumentError: Package PlotlyJS does not have Reexport in its dependencies:
- If you have PlotlyJS checked out for development and have
  added Reexport as a dependency but haven't updated your primary
  environment's manifest file, try `Pkg.resolve()`.
- Otherwise you may need to report an issue with PlotlyJS
Stacktrace:
 [1] require(::Module, ::Symbol) at ./loading.jl:830
 [2] include at ./boot.jl:317 [inlined]
 [3] include_relative(::Module, ::String) at ./loading.jl:1038
 [4] include(::Module, ::String) at ./sysimg.jl:29
 [5] top-level scope at none:2
 [6] eval at ./boot.jl:319 [inlined]
 [7] eval(::Expr) at ./client.jl:399
 [8] top-level scope at ./none:3
in expression starting at /home/sebastian/.julia/packages/PlotlyJS/3eSkF/src/PlotlyJS.jl:5
ERROR: Failed to precompile PlotlyJS [f0f68f2c-4968-5e81-91da-67840de0976a] to /home/sebastian/.julia/compiled/v0.7/PlotlyJS/1r9Ld.ji.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] macro expansion at ./logging.jl:313 [inlined]
 [3] compilecache(::Base.PkgId, ::String) at ./loading.jl:1185
 [4] macro expansion at ./logging.jl:311 [inlined]
 [5] _require(::Base.PkgId) at ./loading.jl:941
 [6] require(::Base.PkgId) at ./loading.jl:852
 [7] macro expansion at ./logging.jl:311 [inlined]
 [8] require(::Module, ::Symbol) at ./loading.jl:834
 [9] top-level scope at /home/sebastian/.julia/packages/Plots/ex9On/src/backends.jl:320
 [10] eval at ./boot.jl:319 [inlined]
 [11] _initialize_backend(::Plots.PlotlyJSBackend) at /home/sebastian/.julia/packages/Plots/ex9On/src/backends.jl:319
 [12] backend(::Plots.PlotlyJSBackend) at /home/sebastian/.julia/packages/Plots/ex9On/src/backends.jl:196
 [13] backend(::Symbol) at /home/sebastian/.julia/packages/Plots/ex9On/src/backends.jl:209
 [14] plotlyjs() at /home/sebastian/.julia/packages/Plots/ex9On/src/backends.jl:29
 [15] top-level scope at none:0
```

**Comment by pdimens** [comment]
I have a similar issue when calling `using PlotlyJS`

```
[ Info: Precompiling PlotlyJS [f0f68f2c-4968-5e81-91da-67840de0976a]
┌ Warning: Package PlotlyJS does not have Reexport in its dependencies:
│ - If you have PlotlyJS checked out for development and have
│   added Reexport as a dependency but haven't updated your primary
│   environment's manifest file, try `Pkg.resolve()`.
│ - Otherwise you may need to report an issue with PlotlyJS
└ Loading Reexport into PlotlyJS from project dependency, future warnings for PlotlyJS are suppressed.
WARNING: could not import Base.quit into AtomShell
ERROR: LoadError: LoadError: UndefVarError: Associative not defined
Stacktrace:
 [1] top-level scope at /home/pdimens/.julia/packages/PlotlyJS/3eSkF/src/display.jl:101
 [2] top-level scope at /home/pdimens/.julia/packages/PlotlyJS/3eSkF/src/display.jl:90
 [3] include at ./boot.jl:317 [inlined]
 [4] include_relative(::Module, ::String) at ./loading.jl:1038
 [5] include at ./sysimg.jl:29 [inlined]
 [6] include(::String) at /home/pdimens/.julia/packages/PlotlyJS/3eSkF/src/PlotlyJS.jl:3
 [7] top-level scope at none:0
 [8] top-level scope at none:2
in expression starting at /home/pdimens/.julia/packages/PlotlyJS/3eSkF/src/display.jl:89
in expression starting at /home/pdimens/.julia/packages/PlotlyJS/3eSkF/src/PlotlyJS.jl:34
ERROR: Failed to precompile PlotlyJS [f0f68f2c-4968-5e81-91da-67840de0976a] to /home/pdimens/.julia/compiled/v1.0/PlotlyJS/1r9Ld.ji.
```

**Comment by sglyon** [comment]
What versions of PlotlyJS and Julia are you trying to use?

**Comment by swiesend** [comment]
Sorry for not having passed this information in first place...

When the error occurred "Reexport" wasn't installed yet.

```
$ julia
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: https://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.7.0 (2018-08-08 06:46 UTC)
 _/ |\__'_|_|_|\__'_|  |  Official http://julialang.org/ release
|__/                   |  x86_64-pc-linux-gnu

(v0.7) pkg> status
    Status `~/.julia/environments/v0.7/Project.toml`
  [c52e3926] Atom v0.7.5
  [336ed68f] CSV v0.3.1
  [aaaa29a8] Clustering v0.12.0
  [944b1d66] CodecZlib v0.5.0
  [8f4d0f93] Conda v1.0.1
  [3a865a2d] CuArrays v0.7.3
  [a93c6f00] DataFrames v0.13.1
  [864edb3b] DataStructures v0.11.1
  [b4f34e82] Distances v0.7.3
  [31c24e10] Distributions v0.16.3
  [587475ba] Flux v0.6.6
  [f6369f11] ForwardDiff v0.9.0
  [c27321d9] Glob v1.2.0
  [7073ff75] IJulia v1.10.0
  [c8e1da08] IterTools v1.0.0
  [e5e0dc1b] Juno v0.5.2
  [b8a86587] NearestNeighbors v0.4.1
  [58dd65bb] Plotly v0.1.1
  [f0f68f2c] PlotlyJS v0.10.3
  [91a5bcdd] Plots v0.19.3
  [438e738f] PyCall v1.18.3
  [189a3867] Reexport v0.2.0
  [2913bbd2] StatsBase v0.25.0
```

But even with Reexport installed I get the same issue:
```
julia> using PlotlyJS
[ Info: Precompiling PlotlyJS [f0f68f2c-4968-5e81-91da-67840de0976a]
┌ Warning: Package PlotlyJS does not have Reexport in its dependencies:
│ - If you have PlotlyJS checked out for development and have
│   added Reexport as a dependency but haven't updated your primary
│   environment's manifest file, try `Pkg.resolve()`.
│ - Otherwise you may need to report an issue with PlotlyJS
└ Loading Reexport into PlotlyJS from project dependency, future warnings for PlotlyJS are suppressed.
WARNING: importing deprecated binding Base.Associative into PlotlyJS.
WARNING: Base.Associative is deprecated, use AbstractDict instead.
  likely near /home/sebastian/.julia/packages/PlotlyJS/3eSkF/src/display.jl:89
WARNING: Base.Associative is deprecated, use AbstractDict instead.
  likely near /home/sebastian/.julia/packages/PlotlyJS/3eSkF/src/display.jl:89
WARNING: Base.Associative is deprecated, use AbstractDict instead.
  likely near /home/sebastian/.julia/packages/PlotlyJS/3eSkF/src/display.jl:89
WARNING: Base.Associative is deprecated, use AbstractDict instead.
  likely near /home/sebastian/.julia/packages/PlotlyJS/3eSkF/src/display.jl:89
┌ Warning: Requires now needs a UUID; please see the readme for changes in 0.7.
└ @ PlotlyJS ~/.julia/packages/PlotlyJS/3eSkF/src/display.jl:123
WARNING: Base.Random is deprecated, run `using Random` instead
  likely near /home/sebastian/.julia/packages/PlotlyJS/3eSkF/src/displays/electron.jl:5
ERROR: LoadError: LoadError: LoadError: UndefVarError: UUID not defined
Stacktrace:
 [1] getproperty(::Module, ::Symbol) at ./sysimg.jl:13
 [2] top-level scope at none:0
 [3] include at ./boot.jl:317 [inlined]
 [4] include_relative(::Module, ::String) at ./loading.jl:1038
 [5] include at ./sysimg.jl:29 [inlined]
 [6] include(::String) at /home/sebastian/.julia/packages/PlotlyJS/3eSkF/src/PlotlyJS.jl:3
 [7] top-level scope at none:0
 [8] include at ./boot.jl:317 [inlined]
 [9] include_relative(::Module, ::String) at ./loading.jl:1038
 [10] include at ./sysimg.jl:29 [inlined]
 [11] include(::String) at /home/sebastian/.julia/packages/PlotlyJS/3eSkF/src/PlotlyJS.jl:3
 [12] top-level scope at none:0
 [13] include at ./boot.jl:317 [inlined]
 [14] include_relative(::Module, ::String) at ./loading.jl:1038
 [15] include(::Module, ::String) at ./sysimg.jl:29
 [16] top-level scope at none:2
 [17] eval at ./boot.jl:319 [inlined]
 [18] eval(::Expr) at ./client.jl:399
 [19] top-level scope at ./none:3
in expression starting at /home/sebastian/.julia/packages/PlotlyJS/3eSkF/src/displays/electron.jl:5
in expression starting at /home/sebastian/.julia/packages/PlotlyJS/3eSkF/src/display.jl:124
in expression starting at /home/sebastian/.julia/packages/PlotlyJS/3eSkF/src/PlotlyJS.jl:34
ERROR: Failed to precompile PlotlyJS [f0f68f2c-4968-5e81-91da-67840de0976a] to /home/sebastian/.julia/compiled/v0.7/PlotlyJS/1r9Ld.ji.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] macro expansion at ./logging.jl:313 [inlined]
 [3] compilecache(::Base.PkgId, ::String) at ./loading.jl:1185
 [4] _require(::Base.PkgId) at ./logging.jl:311
 [5] require(::Base.PkgId) at ./loading.jl:852
 [6] macro expansion at ./logging.jl:311 [inlined]
 [7] require(::Module, ::Symbol) at ./loading.jl:834
```

**Comment by sglyon** [comment]
No problem!

Can you show the output of running the `status` command from the pkg prompt? I would like to see the version of PlotlyJS that you are using

**Comment by swiesend** [comment]
it's `[f0f68f2c] PlotlyJS v0.10.3`

```
(v0.7) pkg> update

(v0.7) pkg> status
    Status `~/.julia/environments/v0.7/Project.toml`
  [c52e3926] Atom v0.7.5
  [336ed68f] CSV v0.3.1
  [aaaa29a8] Clustering v0.12.0
  [944b1d66] CodecZlib v0.5.0
  [8f4d0f93] Conda v1.0.1
  [3a865a2d] CuArrays v0.7.3
  [a93c6f00] DataFrames v0.13.1
  [864edb3b] DataStructures v0.11.1
  [b4f34e82] Distances v0.7.3
  [31c24e10] Distributions v0.16.3
  [587475ba] Flux v0.6.6
  [f6369f11] ForwardDiff v0.9.0
  [c27321d9] Glob v1.2.0
  [7073ff75] IJulia v1.11.0
  [c8e1da08] IterTools v1.0.0
  [e5e0dc1b] Juno v0.5.2
  [b8a86587] NearestNeighbors v0.4.1
  [58dd65bb] Plotly v0.1.1
  [f0f68f2c] PlotlyJS v0.10.3
  [91a5bcdd] Plots v0.19.3
  [438e738f] PyCall v1.18.4
  [189a3867] Reexport v0.2.0
  [2913bbd2] StatsBase v0.25.0
```

```
julia> using PlotlyJS
[ Info: Precompiling PlotlyJS [f0f68f2c-4968-5e81-91da-67840de0976a]
┌ Warning: Package PlotlyJS does not have Reexport in its dependencies:
│ - If you have PlotlyJS checked out for development and have
│   added Reexport as a dependency but haven't updated your primary
│   environment's manifest file, try `Pkg.resolve()`.
│ - Otherwise you may need to report an issue with PlotlyJS
└ Loading Reexport into PlotlyJS from project dependency, future warnings for PlotlyJS are suppressed.
WARNING: importing deprecated binding Base.Associative into PlotlyJS.
WARNING: Base.Associative is deprecated, use AbstractDict instead.
  likely near /home/sebastian/.julia/packages/PlotlyJS/3eSkF/src/display.jl:89
WARNING: Base.Associative is deprecated, use AbstractDict instead.
  likely near /home/sebastian/.julia/packages/PlotlyJS/3eSkF/src/display.jl:89
WARNING: Base.Associative is deprecated, use AbstractDict instead.
  likely near /home/sebastian/.julia/packages/PlotlyJS/3eSkF/src/display.jl:89
WARNING: Base.Associative is deprecated, use AbstractDict instead.
  likely near /home/sebastian/.julia/packages/PlotlyJS/3eSkF/src/display.jl:89
┌ Warning: Requires now needs a UUID; please see the readme for changes in 0.7.
└ @ PlotlyJS ~/.julia/packages/PlotlyJS/3eSkF/src/display.jl:123
WARNING: Base.Random is deprecated, run `using Random` instead
  likely near /home/sebastian/.julia/packages/PlotlyJS/3eSkF/src/displays/electron.jl:5
ERROR: LoadError: LoadError: LoadError: UndefVarError: UUID not defined
Stacktrace:
 [1] getproperty(::Module, ::Symbol) at ./sysimg.jl:13
 [2] top-level scope at none:0
 [3] include at ./boot.jl:317 [inlined]
 [4] include_relative(::Module, ::String) at ./loading.jl:1038
 [5] include at ./sysimg.jl:29 [inlined]
 [6] include(::String) at /home/sebastian/.julia/packages/PlotlyJS/3eSkF/src/PlotlyJS.jl:3
 [7] top-level scope at none:0
 [8] include at ./boot.jl:317 [inlined]
 [9] include_relative(::Module, ::String) at ./loading.jl:1038
 [10] include at ./sysimg.jl:29 [inlined]
 [11] include(::String) at /home/sebastian/.julia/packages/PlotlyJS/3eSkF/src/PlotlyJS.jl:3
 [12] top-level scope at none:0
 [13] include at ./boot.jl:317 [inlined]
 [14] include_relative(::Module, ::String) at ./loading.jl:1038
 [15] include(::Module, ::String) at ./sysimg.jl:29
 [16] top-level scope at none:2
 [17] eval at ./boot.jl:319 [inlined]
 [18] eval(::Expr) at ./client.jl:399
 [19] top-level scope at ./none:3
in expression starting at /home/sebastian/.julia/packages/PlotlyJS/3eSkF/src/displays/electron.jl:5
in expression starting at /home/sebastian/.julia/packages/PlotlyJS/3eSkF/src/display.jl:124
in expression starting at /home/sebastian/.julia/packages/PlotlyJS/3eSkF/src/PlotlyJS.jl:34
ERROR: Failed to precompile PlotlyJS [f0f68f2c-4968-5e81-91da-67840de0976a] to /home/sebastian/.julia/compiled/v0.7/PlotlyJS/1r9Ld.ji.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] macro expansion at ./logging.jl:313 [inlined]
 [3] compilecache(::Base.PkgId, ::String) at ./loading.jl:1185
 [4] _require(::Base.PkgId) at ./logging.jl:311
 [5] require(::Base.PkgId) at ./loading.jl:852
 [6] macro expansion at ./logging.jl:311 [inlined]
 [7] require(::Module, ::Symbol) at ./loading.jl:834
```

**Comment by sglyon** [comment]
Ahh that's the problem.

The currently released version of PlotlyJS does not work with julia 0.7

But, the master branch does. To try it out run `add PlotlyJS#master` from the package prompt

We are waiting on some METADATA.jl pull requests to be merged before we can make a release of this package.

**Comment by pdimens** [comment]
I'm using Julia 1.0 (on Arch Linux 64bit if that matters) and the this is the output of `status` : 
```
(v1.0) pkg> status
    Status `~/.julia/environments/v1.0/Project.toml`
  [7d9fca2a] Arpack v0.2.3
  [c52e3926] Atom v0.7.5
  [7e6ae17a] BioSequences v1.0.0
  [ad839575] Blink v0.8.0
  [a1bb12fb] Electron v0.2.0
  [28b8d3ca] GR v0.32.3
  [c91e804a] Gadfly v0.8.0
  [e5e0dc1b] Juno v0.5.2
  [a975b10e] Mux v0.5.3
  [4afe4657] Phylogenetics v0.0.2
  [58dd65bb] Plotly v0.1.1
  [a03496cd] PlotlyBase v0.2.1
  [f0f68f2c] PlotlyJS v0.10.3
  [91a5bcdd] Plots v0.20.1
  [d330b81b] PyPlot v2.6.2
  [ce6b1742] RDatasets v0.5.0
  [189a3867] Reexport v0.2.0
  [60ddc479] StatPlots v0.8.1
  [0f1e0344] WebIO v0.3.2
```

**Comment by pdimens** [comment]
using the non-master branch, PlotlyJS is updated to v0.10.3 today and fails to compile with this error:
```
ERROR: LoadError: LoadError: UndefVarError: Associative not defined
Stacktrace:
 [1] top-level scope at /home/pdimens/.julia/packages/PlotlyJS/3eSkF/src/display.jl:101
 [2] top-level scope at /home/pdimens/.julia/packages/PlotlyJS/3eSkF/src/display.jl:90
 [3] include at ./boot.jl:317 [inlined]
 [4] include_relative(::Module, ::String) at ./loading.jl:1038
 [5] include at ./sysimg.jl:29 [inlined]
 [6] include(::String) at /home/pdimens/.julia/packages/PlotlyJS/3eSkF/src/PlotlyJS.jl:3
 [7] top-level scope at none:0
 [8] top-level scope at none:2
in expression starting at /home/pdimens/.julia/packages/PlotlyJS/3eSkF/src/display.jl:89
in expression starting at /home/pdimens/.julia/packages/PlotlyJS/3eSkF/src/PlotlyJS.jl:34
```

**Comment by swiesend** [comment]
checking out the master `[f0f68f2c] PlotlyJS v0.10.3+ #master (https://github.com/sglyon/PlotlyJS.jl.git)` helped. But unfortunately Plots.jl still can't handle it.

**Comment by sglyon** [comment]
Version 0.12.0 of this package has been released and is compatible with Julia 1.0
